flyteidl/go.mod

```diff
@@ -7,9 +7,11 @@ toolchain go1.21.3
 require (
 	github.com/flyteorg/flyte/flytestdlib v0.0.0-00010101000000-000000000000
 	github.com/go-test/deep v1.0.7
+	github.com/golang/glog v1.2.0
 	github.com/golang/protobuf v1.5.3
 	github.com/grpc-ecosystem/go-grpc-middleware v1.1.0
 	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0
+	github.com/grpc-ecosystem/grpc-gateway v1.16.0
```

Comment:
why are we changing this?

Reply:
`make generate` in flyteidl requested to run `go mod tidy`:

```console
~ make generate
go generate ./...
go: updates to go.mod needed; to update it:
        go mod tidy
make: *** [Makefile:13: generate] Error 1
```

After rebasing and re-generating the protos now, these further changes occured:
https://github.com/flyteorg/flyte/pull/4726/commits/683a6a550bb5ab98a88f14e499bc9c6144da1844